### PR TITLE
Updating the prowjob CRD to v1

### PR DESCRIPTION
**Pull request by bradmwilliams**
The prowjob api has been available at `v1` for quite some time.  This PR bumps the existing CRDs from `v1beta1` to `v1`.

### CODE REVIEW

**Review by alvaroaleman — COMMENTED** [review]
/cc @chaodaiG 
At what kube version is the prow.k8s.io cluster? This need 1.16+ : https://www.openshift.com/blog/a-look-into-the-technical-details-of-kubernetes-1-16

:upside_down_face:

**Comment by chaodaiG** [comment]
It's already 1.16+, all our prow instances as well, as prometheus operator wants 1.16+ too. 
/cc @fejta 
Who is oncall

**Review by alvaroaleman — COMMENTED** [review]
/ok-to-test

**Comment by bradmwilliams** [comment]
I've signed the CLA

File: config/prow-staging/cluster/prowjob_customresourcedefinition.yaml
Comment (alvaroaleman) [inline config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:12]: @bradmwilliams you also need to change:
* config/prow/cluster/starter-gcs.yaml
* config/prow/cluster/starter-s3.yaml
* prow/test/integration/prow/cluster/100_starter.yaml (most important one as used by integration tests)

Comment (bradmwilliams) [reply config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:12]: @alvaroaleman The `starter-*.yaml` files have been accounted for.  Looking at the third file now.

Comment (bradmwilliams) [reply config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:12]: Done!

**Comment by alvaroaleman** [comment]
/retest

Weird, it fails waiting for sinker to delete the pod and prowjob, but sinker does that, according to the log

**Comment by bradmwilliams** [comment]
@alvaroaleman It appears that the issue is the pod is not getting flagged for deletion before the 60 second time limit expires.

**Comment by chaodaiG** [comment]
/hold
I would be concerned if the integration test passed with so many retries

**Comment by chaodaiG** [comment]
Trying to get you unblocked, sorry for being late here. I'm able to reproduce this locally, for some reason sinker is not cleaning the pods with your change, I'll take a deeper look to see what's going on.

What's weird is that sinker logs says it's deleting these pods, but nothing really happened

**Comment by chaodaiG** [comment]
There were several weird things happening in my local testing:
- Added a log at https://github.com/kubernetes/test-infra/blob/70fe6633179dedad1576229bee7d75a78e5f0dcc/prow/cmd/sinker/main.go#L506, and it always returns `k8serrors.IsNotFound(err)` for all test cases, including  `running-pod`, `orphaned-pod` etc.
- The prowjobs that were not found by sinker were discoverable by `kubectl`.

Compared prow jobs logs:
- The one failed in this PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21805/pull-test-infra-integration/1384245481613299712
- One passed in other PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/21875/pull-test-infra-integration/1384681190530748416

And the sinker logs from the artifacts of the spyglass links above:
- https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/21875/pull-test-infra-integration/1384681190530748416/artifacts/prow_pod_logs/
- https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/test-infra/21805/pull-test-infra-integration/1384245481613299712/artifacts/prow_pod_logs/

One noticeable difference is that this log line:
```
{"component":"sinker","file":"prow/cmd/sinker/main.go:324","func":"main.(*controller).clean","job":"","level":"info","msg":"Deleted prowjob.","name":"max-prow-job-age-0dfcc0aebfc0b5e6adb2e4aa72ed9eb8","severity":"info","time":"2021-04-21T01:41:20Z","type":""}
```
Only exists in the passed run but not in this PR, feels like sinker is not able to delete prowjob that exceeds max prow job age.

File: config/prow-staging/cluster/prowjob_customresourcedefinition.yaml
Comment (chaodaiG) [inline config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:50]:
```suggestion
            type: object
            x-kubernetes-preserve-unknown-fields: true
```

Comment (chaodaiG) [reply config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:50]: From my local testing I noticed that `prowjob` CR created with PR only has `state` field under `status`, where a lot more fields are expected such as `startTime` etc. which very likely is the reason why the tests failed. Adding `x-kubernetes-preserve-unknown-fields: true` passed sinker tests at least. 

We will need to be more careful though on this change as this might need to be set elsewhere too.
@cjwagner  and @alvaroaleman  for awareness

**Review by chaodaiG — CHANGES_REQUESTED** [review]

File: config/prow-staging/cluster/prowjob_customresourcedefinition.yaml
Comment (alvaroaleman) [reply config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:50]: Oh ofc :facepalm: We also need the same for spec, what we have there currently is definitely not exhaustive.

I also tried to generate a complete schema for this at some point, but hit some roadblocks and didn't pursue it further: https://github.com/kubernetes/test-infra/pull/19522

File: config/prow-staging/cluster/prowjob_customresourcedefinition.yaml
Comment (alvaroaleman) [inline config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:51]: This is also needed for spec

Comment (bradmwilliams) [reply config/prow-staging/cluster/prowjob_customresourcedefinition.yaml:51]: ack

**Review by alvaroaleman — APPROVED** [review]
/label tide/merge-method-squash
/hold cancel
Chao will need to approve though, I am not a root approver

**Comment by chaodaiG** [comment]
LGTM, would defer to @cjwagner who is oncall

**Comment by cjwagner** [comment]
LGTM as well, but I'll hold off on merging this until Monday in case we encounter unexpected issues.

**Review by cjwagner — APPROVED** [review]

**Pull request merged**
